exec/rpc/src/main/java/org/apache/drill/exec/rpc/AbstractRemoteConnection.java

```diff
@@ -224,4 +237,67 @@ public void close() {
     }
   }
 
+  /**
+   * Helps to add all the required security handler's after negotiation for encryption is completed.
+   * <p>
+   *  Handler's that are added are:
+   *    <li>  SaslDecryptionHandler
+   *    <li>  LengthFieldBasedFrameDecoder Handler
+   *    <li>  SaslEncryptionHandler
+   *    <li>  ChunkCreationHandler
+   * </p>
+   * <p>
+   *  If encryption is enabled ChunkCreationHandler is always added irrespective of chunkMode enabled or not.
+   *  This helps to make a generic encryption handler.
+   * </p>
+   */
+  @Override
+  public void addSecurityHandlers() {
+
+    final ChannelPipeline channelPipeline = getChannel().pipeline();
+    channelPipeline.addFirst("SaslDecryptionHandler", new SaslDecryptionHandler(saslBackend, getWrappedChunkSize(),
```

Comment:
Define handler names as class constants, specially "message-decoder", which is from another class.